### PR TITLE
backend: testSessionByTeacherId Resolver and Query

### PR DESCRIPTION
**Pull request by wais-hundekar**
## Notion ticket link

Update Backend to Fetch Test Sessions by Teacher](https://www.notion.so/uwblueprintexecs/Update-backend-to-fetch-test-sessions-by-teacher-5e3c9d2e2ef649759390dd61b7fe715e)

## Implementation description
* Implemented Resolver and Query for testSessionByTeacherId
Screenshot 2023-05-01 at 1 14 02 AM
Screenshot 2023-05-01 at 7 54 00 PM

## Steps to test
1. Test query with command in screenshot

## What should reviewers focus on?
* Ensure all types are correct
* Ensure that query return is also correct

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/graphql/resolvers/testSessionResolvers.ts [hunk 1]
```diff
@@ -59,6 +59,12 @@ const testSessionResolvers = {
     ): Promise<TestSessionResponseDTO> => {
       return testSessionService.getTestSessionByAccessCode(accessCode);
     },
+    testSessionByTeacherId: async (
```
Comment (carissa-tang) [inline backend/graphql/resolvers/testSessionResolvers.ts]:
```suggestion
    testSessionsByTeacherId: async (
```

**Review by carissa-tang — COMMENTED** [review]
can you test what happens if you call this query on a teacher with 0 test sessions? would be great if you can add a unit test for that case. you might also need to update the return type since might be []

**Comment by wais-hundekar** [comment]
> can you test what happens if you call this query on a teacher with 0 test sessions? would be great if you can add a unit test for that case. you might also need to update the return type since might be []

@carissa-tang What would be the difference between this and the existing "getTestSessionsByTeacherId for invalid teacher id" test? If the function works by searching through all the test sessions to check if the teacherId matches for any of them, wouldn't the invalid id case also be the same result as the case where no matching teacher id exists (no test session for a teacher?)

File: frontend/src/APIClients/queries/TestSessionQueries.ts [hunk 2]
```diff
@@ -11,4 +11,13 @@ const GET_TEST_SESSION_BY_ACCESS_CODE = gql`
   }
 `;
 
-export default GET_TEST_SESSION_BY_ACCESS_CODE;
+export const GET_TEST_SESSION_BY_TEACHER_ID = gql`
+  query TestSessionByTeacherId($teacherId: String!) {
+    testSessionByTeacherId(teacherId: $teacherId) {
```
Comment (carissa-tang) [inline frontend/src/APIClients/queries/TestSessionQueries.ts]:
```suggestion
    testSessionsByTeacherId(teacherId: $teacherId) {
```

**Review by carissa-tang — APPROVED** [review]

**Pull request merged**